watertap-org/watertap
# ADM1

## Fixes/Resolves:

Implement Anaerobic Digestion Model No. 1 (ADM1) for detailed wastewater treatment processes.

Replaces #627 

Checklist:

- [x] add Petersen matrix values
- [x]  add process rate equations
- [x]  add inhibition functions, which are included in process rate equations
- [x]  verify Petersen matrix values
- Add quasi-steady-state components and related constraints and verify for a known state
- [x]  Add quasi-steady-state components 
- [x] Verify unit conversion (kg/mol)
- [x] Retest
- If possible without VLE piece, verify rate equations 
- [x]  Calculate total H2, CO2 and CH4
- [x] Retest in test_adm1_reactions.py
- Add the necessary properties for VLE to the thermo package
- [x]  Make vapor property package
- Build the anaerobic digestor unit model and verify the whole system
- [x] Create reactor model
- [x] Move constraint from flowsheet 
- [x] Retest


## Summary/Motivation:


## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.
